control/controlhttp/client.go

```diff
@@ -96,17 +97,50 @@ func (a *Dialer) httpsFallbackDelay() time.Duration {
 }
 
 var _ = envknob.RegisterBool("TS_USE_CONTROL_DIAL_PLAN") // to record at init time whether it's in use
+var _ = envknob.RegisterBool("TS_USE_SRV_DIAL_PLAN")
 
 func (a *Dialer) dial(ctx context.Context) (*ClientConn, error) {
-	// If we don't have a dial plan, just fall back to dialing the single
-	// host we know about.
-	useDialPlan := envknob.BoolDefaultTrue("TS_USE_CONTROL_DIAL_PLAN")
-	if !useDialPlan || a.DialPlan == nil || len(a.DialPlan.Candidates) == 0 {
-		return a.dialHost(ctx, netip.Addr{})
+	useCtrlDialPlan := envknob.BoolDefaultTrue("TS_USE_CONTROL_DIAL_PLAN")
+	useSRVDialPlan := envknob.BoolDefaultTrue("TS_USE_SRV_DIAL_PLAN")
+
+	candidates := []tailcfg.ControlIPCandidate{}
+	if useCtrlDialPlan && a.DialPlan != nil && len(a.DialPlan.Candidates) > 0 {
+		candidates = a.DialPlan.Candidates
+	} else {
+		candidates = []tailcfg.ControlIPCandidate{
+			tailcfg.ControlIPCandidate{},
+		}
 	}
-	candidates := a.DialPlan.Candidates
 
-	// Otherwise, we try dialing per the plan. Store the highest priority
+	if useSRVDialPlan {
+		resolver := &net.Resolver{
+			PreferGo: true, // Use the Go resolver implementation
+		}
+		// Perform a DNS lookup using the custom resolver
+		_, srvRecords, err := resolver.LookupSRV(ctx, "ts2021", "tcp", a.Hostname)
```

Comment:
We don't want the client to be doing these new SRV DNS lookups by default for most users using the default URL which doesn't need this.

Maybe we need to encode the use-SRV-or-not condition (which you currently have as an envknob, defaulting to true) to instead be part of the login server URL.

Like perhaps an explicit port number of `0` means to use SRV.

Reply:
i've been using apache hive at work, and they support a bunch of different "transports."  they prefix them to the schema like this `jdbc:hive2://hive:10000`
does it make sense to do something like that in this case?, Explicit port 0 would work without any assumptions, or prefixing the schema would be extensible.
srv:http://control.plane.net
Just brainstorming, I'm open to anything, really.

Reply:
I'm afraid that a prefix (or anything that's not shaped like a valid URL) would cause additional complications.

I'd prefer either port 0 (which is effectively unusable with typical OSes, even if it's valid on the wire) or a fragment or URL parameter in the URL, like `https://foo.bar.com:0` or `https://foo.bar.com#srv` or `https://foo.bar.com?tsport=srv`. The port 0 thing is the most succinct, so that's kinda my preference.

Reply:
@bradfitz yeah those are better ideas.. I prefer the fragment, if only because once we use 0 we've shot our shot.  god forbid some other dial nonsense come up, other fragments could be used.
but, frankly, I'm just grateful your considering it. you pick and I'll implement.